app/models/host/managed.rb

```diff
@@ -33,6 +32,8 @@ def self.complete_for(query, opts = {})
   define_model_callbacks :build, :only => :after
   define_model_callbacks :provision, :only => :before
 
+  before_validation :refresh_build_status, :if => :build_changed?
```

Comment:
should be covered here https://github.com/theforeman/foreman/pull/2580/files#diff-88c8d0302188366e58cfd358acf05756R437